core/src/Kokkos_View.hpp

```diff
@@ -2066,7 +2162,20 @@ class View : public ViewTraits<DataType, Properties...> {
   }
   template <class Traits>
   KOKKOS_INLINE_FUNCTION View(
-      const track_type& track,
+      const view_tracker_type& track,
+      const Kokkos::Impl::ViewMapping<Traits, typename Traits::specialize>& map)
+      : m_track(track), m_map() {
+    typedef Kokkos::Impl::ViewMapping<traits, Traits,
+                                      typename traits::specialize>
+        Mapping;
+    static_assert(Mapping::is_assignable,
+                  "Incompatible View copy construction");
+    Mapping::assign(m_map, map, track.m_tracker);
+  }
+
+  template <class Traits>
+  KOKKOS_INLINE_FUNCTION View(
+      const typename view_tracker_type::track_type& track,
```

Comment:
It's not clear why these are separate constructors. Can you add a comment clarifying why we need both of these, even though they do the exact same thing?

Also, if they do the exact same thing, one should forward to the other; I still prefer to avoid copying code when you can

Reply:
I added some comments clarifying this.  the view_tracker_type::track_type& is only for backward compatibility.  Still not convinced that they should be converged.